### ISSUE TITLE
Please add support for recent versions of x509 et al

### ISSUE THREAD

**Issue by peti** [derekelkins/sws#10]
http://packdeps.haskellers.com/feed?needle=sws

**Comment by derekelkins** [comment]
This won't happen until x509 recovers the functionality it use to have.

**Comment by Ornedan** [comment]
This renders sws uninstallable on Haskell Platform 7.10.2-a (probably also 7.10.2). On a clean installation:

```
$ cabal install sws
Resolving dependencies...
In order, the following would be installed:
ansi-terminal-0.6.2.2 (via: wai-extra-3.0.10) (new package)
appar-0.1.4 (via: iproute-1.5.0) (new package)
auto-update-0.1.2.2 (via: warp-3.0.13.1 wai-logger-2.2.4.1 fast-logger-2.4.1) (new package)
base64-bytestring-1.0.0.1 (via: wai-extra-3.0.10 pem-0.2.2) (new package)
blaze-builder-0.4.0.1 (via: wai-extra-3.0.10 warp-3.0.13.1 wai-logger-2.2.4.1 wai-3.0.3.0 streaming-commons-0.1.12.1 http-types-0.8.6 cookie-0.4.1.6) (new package)
byteable-0.1.1 (via: tls-1.2.18 x509-validation-1.5.2 crypto-pubkey-0.2.8 cprng-aes-0.6.1 cipher-rc4-0.1.4 cipher-des-0.0.6 cipher-aes-0.2.11 crypto-cipher-types-0.0.9 securemem-0.1.9 cryptohash-0.11.6) (new package)
byteorder-1.0.4 (via: wai-logger-2.2.4.1 iproute-1.5.0) (new package)
bytestring-builder-0.10.6.0.0 (via: fast-logger-2.4.1) (new package)
cereal-0.4.1.1 (via: tls-1.2.18) (new package)
cryptohash-0.11.6 (via: tls-1.2.18 x509-validation-1.5.2 x509-1.5.1 crypto-pubkey-0.2.8) (new package)
data-default-class-0.0.1 (via: warp-tls-3.0.4.2 tls-1.2.18 x509-validation-1.5.2 wai-extra-3.0.10 cookie-0.4.1.6) (new package)
cookie-0.4.1.6 (via: wai-extra-3.0.10) (new package)
filepath-1.3.0.1 (latest: 1.4.0.0) (via: sws-0.3.1.1 x509-validation-1.5.2 x509-store-1.5.0 x509-1.5.1 wai-middleware-static-0.6.0.1 process-1.2.3.0 fast-logger-2.4.1 easy-file-0.2.1 directory-1.2.3.1) (new version)
directory-1.2.3.1 (via: sws-0.3.1.1 x509-validation-1.5.2 x509-store-1.5.0 x509-1.5.1 wai-extra-3.0.10 wai-middleware-static-0.6.0.1 streaming-commons-0.1.12.1 process-1.2.3.0 fast-logger-2.4.1 easy-file-0.2.1) (new version)
easy-file-0.2.1 (via: wai-logger-2.2.4.1) (new package)
fast-logger-2.4.1 (via: wai-extra-3.0.10 wai-logger-2.2.4.1) (new package)
hourglass-0.2.9 (via: sws-0.3.1.1 x509-validation-1.5.2 x509-1.5.1 asn1-encoding-0.9.0 asn1-types-0.3.0) (new package)
asn1-types-0.3.0 (via: sws-0.3.1.1 tls-1.2.18 x509-validation-1.5.2 x509-store-1.5.0 x509-1.5.1 crypto-pubkey-types-0.4.3 asn1-parse-0.9.1 asn1-encoding-0.9.0) (new package)
asn1-encoding-0.9.0 (via: sws-0.3.1.1 tls-1.2.18 x509-validation-1.5.2 x509-store-1.5.0 x509-1.5.1 crypto-pubkey-types-0.4.3 asn1-parse-0.9.1) (new package)
asn1-parse-0.9.1 (via: x509-1.5.1) (new package)
crypto-pubkey-types-0.4.3 (via: tls-1.2.18 x509-validation-1.5.2 x509-store-1.5.0 x509-1.5.1 crypto-pubkey-0.2.8) (new package)
http-date-0.0.6.1 (via: warp-3.0.13.1) (new package)
http-types-0.8.6 (via: sws-0.3.1.1 wai-extra-3.0.10 warp-3.0.13.1 wai-middleware-static-0.6.0.1 wai-logger-2.2.4.1 wai-3.0.3.0) (new package)
iproute-1.5.0 (via: warp-3.0.13.1) (new package)
memory-0.8 (via: securemem-0.1.9) (new package)
mmorph-1.0.4 (via: resourcet-1.1.6) (new package)
nats-1 (via: semigroups-0.16.2.2) (new package)
pem-0.2.2 (via: sws-0.3.1.1 x509-validation-1.5.2 x509-store-1.5.0 x509-1.5.1) (new package)
process-1.2.3.0 (via: x509-validation-1.5.2 x509-store-1.5.0 x509-1.5.1 streaming-commons-0.1.12.1) (reinstall) (changes: directory-1.2.2.0 -> 1.2.3.1, filepath-1.4.0.0 -> 1.3.0.1)
securemem-0.1.9 (via: crypto-random-0.0.9 cipher-des-0.0.6 cipher-aes-0.2.11 crypto-cipher-types-0.0.9) (new package)
crypto-cipher-types-0.0.9 (via: tls-1.2.18 cipher-rc4-0.1.4 cipher-des-0.0.6 cipher-aes-0.2.11) (new package)
cipher-aes-0.2.11 (via: tls-1.2.18 cprng-aes-0.6.1) (new package)
cipher-des-0.0.6 (via: tls-1.2.18) (new package)
cipher-rc4-0.1.4 (via: tls-1.2.18) (new package)
crypto-random-0.0.9 (via: sws-0.3.1.1 tls-1.2.18 crypto-pubkey-0.2.8 crypto-numbers-0.2.7 cprng-aes-0.6.1) (new package)
cprng-aes-0.6.1 (via: warp-tls-3.0.4.2) (new package)
crypto-numbers-0.2.7 (via: tls-1.2.18 crypto-pubkey-0.2.8) (new package)
crypto-pubkey-0.2.8 (via: sws-0.3.1.1 tls-1.2.18 x509-validation-1.5.2) (new package)
semigroups-0.16.2.2 (via: void-0.7) (new package)
simple-sendfile-0.2.21 (via: warp-3.0.13.1) (new package)
streaming-commons-0.1.12.1 (via: warp-tls-3.0.4.2 wai-extra-3.0.10 warp-3.0.13.1) (new package)
stringsearch-0.3.6.6 (via: wai-extra-3.0.10) (new package)
transformers-base-0.4.4 (via: resourcet-1.1.6 lifted-base-0.2.3.6 monad-control-1.0.0.4) (new package)
monad-control-1.0.0.4 (via: resourcet-1.1.6 lifted-base-0.2.3.6) (new package)
lifted-base-0.2.3.6 (via: wai-extra-3.0.10 resourcet-1.1.6) (new package)
resourcet-1.1.6 (via: sws-0.3.1.1 wai-extra-3.0.10) (new package)
unix-compat-0.4.1.4 (via: wai-extra-3.0.10 warp-3.0.13.1) (new package)
unix-time-0.3.5 (via: wai-logger-2.2.4.1) (new package)
vault-0.3.0.4 (via: wai-extra-3.0.10 warp-3.0.13.1 wai-3.0.3.0) (new package)
void-0.7 (via: wai-extra-3.0.10) (new package)
wai-3.0.3.0 (via: sws-0.3.1.1 warp-tls-3.0.4.2 wai-extra-3.0.10 warp-3.0.13.1 wai-middleware-static-0.6.0.1 wai-logger-2.2.4.1) (new package)
wai-logger-2.2.4.1 (via: wai-extra-3.0.10) (new package)
wai-middleware-static-0.6.0.1 (latest: 0.7.0.1) (via: sws-0.3.1.1) (new package)
warp-3.0.13.1 (latest: 3.1.3) (via: sws-0.3.1.1 warp-tls-3.0.4.2) (new package)
word8-0.1.2 (via: wai-extra-3.0.10) (new package)
wai-extra-3.0.10 (via: sws-0.3.1.1) (new package)
x509-1.5.1 (latest: 1.6.0) (via: sws-0.3.1.1 tls-1.2.18 x509-validation-1.5.2 x509-store-1.5.0) (new package)
x509-store-1.5.0 (latest: 1.6.0) (via: tls-1.2.18 x509-validation-1.5.2) (new package)
x509-validation-1.5.2 (latest: 1.6.0) (via: tls-1.2.18) (new package)
tls-1.2.18 (latest: 1.3.2) (via: warp-tls-3.0.4.2) (new package)
warp-tls-3.0.4.2 (latest: 3.1.2) (via: sws-0.3.1.1) (new package)
sws-0.3.1.1 (new package)
cabal: The following packages are likely to be broken by the reinstalls:
ghc-7.10.2
Cabal-1.22.4.0
Use --force-reinstalls if you want to install anyway.
```

**Comment by Ornedan** [comment]
It looks like you could let directory, filepath, wai and warp to use newer versions, while keeping x509 fixed and it would work.

**Comment by derekelkins** [comment]
@Ornedan Thanks for the heads up.  I pushed a change to raise some upper bounds.  I tested on 7.10.1, so I don't know if it is fine for the Haskell Platform, but it should be.  I'm pretty sure only filepath was the culprit, but other than x509, I raised all bounds to the latest as I doubt they break anything I care about.

As for the x509 situation, vincenthz/hs-certificate#55 is the blocking issue.  Eventually, I'll probably make a pull request to fix it myself if it doesn't get resolved.
